### PR TITLE
Modified the RocketMQ document based on V1.10

### PR DESCRIPTION
**Pull request by jinjianfeng-chn**
Signed-off-by: jinjianfeng-chn <feng081212@163.com>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 1]
```diff
@@ -19,42 +19,64 @@ spec:
...
-| sendTimeOutSec   |    N     | Timeout duration for publishing a message in seconds                                                      | `0`                                |
+| Field                                 | Required | Details                                                      | default                                                 | Example                                      |
+| ------------------------------------- | :------: | ------------------------------------------------------------ | ------------------------------------------------------- | -------------------------------------------- |
+| instanceName                          |    N     | Instance name                                                | time.Now().String()                                     | dapr-rocketmq-test                           |
+| consumerGroup                         |    N     | Consumer group name, it is  recommended. if producerGroup is null，groupName is used. |                                                         | dapr-rocketmq-test-g-c                       |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| consumerGroup                         |    N     | Consumer group name. Recommend. If `producerGroup` is `null`，`groupName` is used. |                                                         | `dapr-rocketmq-test-g-c `                     |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 2]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| producerGroup (consumerID)            |    N     | Producer group Name, it is  recommended. if producerGroup is null，consumerID is used, if consumerID also is null, groupName is used. |                                                         | dapr-rocketmq-test-g-p                       |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| producerGroup (consumerID)            |    N     | Producer group name. Recommended. If `producerGroup` is `null`，`consumerID` is used. If `consumerID` also is null, `groupName` is used. |                                                         | dapr-`rocketmq-test-g-p`                      |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 3]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| groupName                             |    N     | Consumer/Producer group name, it is Deprecated               |                                                         | dapr-rocketmq-test-g                         |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| groupName                             |    N     | Consumer/Producer group name. **Depreciated**.               |                                                         | `dapr-rocketmq-test-g`                         |
```

Comment (hhunter-ms) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]: @msfussell @greenie-msft just want to check - how do we typically handle a depreciated metadata field in the docs. Do we remove it, or ok to mention it like above?

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 4]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| nameSpace                             |    N     | rocketmq namespace                                           |                                                         | dapr-rocketmq                                |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| nameSpace                             |    N     | RocketMQ namespace                                           |                                                         | `dapr-rocketmq`                                |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 5]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| nameServerDomain                      |    N     | rocketmq name server domain                                  |                                                         | https://my-app.net:8080/nsaddr               |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| nameServerDomain                      |    N     | RocketMQ name server domain                                  |                                                         | `https://my-app.net:8080/nsaddr`               |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 6]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| nameServer                            |    N     | rocketmq name server, It's separated by "," or ";"           |                                                         | 127.0.0.1:9876;127.0.0.2:9877,127.0.0.3:9877 |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| nameServer                            |    N     | RocketMQ name server, separated by "," or ";"           |                                                         | 127.0.0.1:9876;127.0.0.2:9877,127.0.0.3:9877 |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 7]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| accessKey                             |    N     | Access Key (Username)                                        |                                                         | `"admin"`                                    |
+| secretKey                             |    N     | Secret Key (Password)                                        |                                                         | `"password"`                                 |
+| securityToken                         |    N     | Security Token                                               |                                                         |                                              |
+| retries                               |    N     | Retry times to send msg to broker                            | 3                                                       | 3                                            |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| retries                               |    N     | Number of retries to send a message to broker                            | 3                                                       | 3                                            |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 8]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| producerQueueSelector (queueSelector) |    N     | Producer Queue selector. There are five implementations of queue selector，hash, random, manual, roundRobin, dapr(default)，respectively. | dapr                                                    | hash                                         |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| producerQueueSelector (queueSelector) |    N     | Producer Queue selector. There are five implementations of queue selector: `hash`, `random`, `manual`, `roundRobin`, `dapr`. | `dapr`                                                    | `hash`                                         |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 9]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| consumerModel                         |    N     | Message model defines the way how messages are delivered to each consumer clients. RocketMQ supports two message models: clustering and broadcasting. | clustering                                              | broadcasting                                 |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| consumerModel                         |    N     | Message model that defines how messages are delivered to each consumer client. RocketMQ supports two message models: `clustering` and `broadcasting`. | `clustering`                                              | `broadcasting`                                 |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 10]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| fromWhere (consumeFromWhere)          |    N     | Consuming point on consumer booting. There are three consuming points: CONSUME_FROM_LAST_OFFSET, CONSUME_FROM_FIRST_OFFSET, CONSUME_FROM_TIMESTAMP | ConsumeFromLastOffset                                   | CONSUME_FROM_LAST_OFFSET                     |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| fromWhere (consumeFromWhere)          |    N     | Consuming point on consumer booting. There are three consuming points: `CONSUME_FROM_LAST_OFFSET`, `CONSUME_FROM_FIRST_OFFSET`, `CONSUME_FROM_TIMESTAMP` | `CONSUME_FROM_LAST_OFFSET`                                   | `CONSUME_FROM_LAST_OFFSET`                     |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 11]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| consumeOrderly                        |    N     | Whether it is an ordered message using FIFO order. This field defaults to false. | false                                                   | false                                        |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| consumeOrderly                        |    N     | Determines if it's an ordered message using FIFO order.  | `false`                                                   | `false`                                        |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 12]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| consumeMessageBatchMaxSize            |    N     | Batch consumption size out of range [1, 1024]                | 512                                                     | 10                                           |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| consumeMessageBatchMaxSize            |    N     | Batch consumption size out of range `[1, 1024]`                | `512`                                                     | `10`                                           |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 13]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| maxReconsumeTimes                     |    N     | Max re-consume times. -1 means 16 times. If messages are re-consumed more than {@link maxReconsumeTimes} before Success, it's be directed to a deletion queue waiting. | orderly message is MaxInt32, concurrently message is 16 | 16                                           |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| maxReconsumeTimes                     |    N     | Max re-consume times. `-1` means 16 times. If messages are re-consumed more than {@link maxReconsumeTimes} before success, they'll be directed to a deletion queue. | Orderly message is `MaxInt32`; Concurrently message is `16` | `16`                                          |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 14]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| autoCommit                            |    N     | auto commit                                                  | true                                                    | false                                        |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| autoCommit                            |    N     | Enable auto commit                                                  | `true`                                                    | `false`                                        |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 15]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| pullInterval                          |    N     | Message pull Interval                                        | 100                                                     | 100                                          |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| pullInterval                          |    N     | Message pull interval                                        | `100`                                                     | `100`                                          |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 16]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| pullBatchSize                         |    N     | The number of messages pulled from the broker at a time. if pullBatchSize is null，ConsumerBatchSize is used. pullBatchSize out of range [1, 1024] | 32                                                      | 10                                           |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| pullBatchSize                         |    N     | The number of messages pulled from the broker at a time. If `pullBatchSize` is `null`, use `ConsumerBatchSize`. `pullBatchSize` out of range `[1, 1024]` | `32`                                                      | `10`                                           |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 17]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| content-type                          |    N     | Message content-type, e.g., `"application/cloudevents+json; charset=utf-8"`, `"application/octet-stream"`. This field defaults to "text/plain" |                                                         | `"text/plain"`                               |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| content-type                          |    N     | Message content type.  | `"text/plain"`                                                        | `"application/cloudevents+json; charset=utf-8"`, `"application/octet-stream"`                              |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 18]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| logLevel                              |    N     | Log level                                                    | warn                                                    | Info                                         |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| logLevel                              |    N     | Log level                                                    | `warn`                                                    | `info`                                         |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 19]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| sendTimeOut                           |    N     | send msg timeout to connect rocketmq's broker, nanoseconds. It is deprecated. | 3 seconds                                               | 10000000000                                  |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| sendTimeOut                           |    N     | Send message timeout to connect RocketMQ's broker, measured in nanoseconds. **Deprecated**. | 3 seconds                                               | `10000000000`                                  |
```

For default, just checking - 3 seconds or 3 nanoseconds?

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 20]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| sendTimeOutSec                        |    N     | Timeout duration for publishing a message in seconds. if sendTimeOutSec is null，sendTimeOut is used. | 3 seconds                                               | 3                                            |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| sendTimeOutSec                        |    N     | Timeout duration for publishing a message in seconds. If `sendTimeOutSec` is `null`, `sendTimeOut` is used. | 3 seconds                                               | `3`                                            |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 21]
```diff
@@ -19,42 +19,64 @@ spec:
...
+| mspProperties                         |    N     | The RocketMQ message properties in this collection are passed to the APP in Data Separate multiple properties with "," |                                                         | key,mkey                                     |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| mspProperties                         |    N     | The RocketMQ message properties in this collection are passed to the APP in Data Separate multiple properties with "," |                                                         | `key,mkey`                                     |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 22]
```diff
@@ -19,42 +19,64 @@ spec:
...
 
 For backwards-compatibility reasons, the following values in the metadata are supported, although their use is discouraged.
 
 | Field (supported but deprecated) | Required | Details                                                  | Example                  |
 | -------------------------------- | :------: | -------------------------------------------------------- | ------------------------ |
 | groupName                        |    N     | Producer group name for RocketMQ publishers              | `"my_unique_group_name"` |
 | sendTimeOut                      |    N     | Timeout duration for publishing a message in nanoseconds | `0`                      |
+| consumerBatchSize                |    N     | The number of messages pulled from the broker at a time  | 32                       |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
| consumerBatchSize                |    N     | The number of messages pulled from the broker at a time  | `32`                       |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 23]
```diff
@@ -79,7 +101,34 @@ curl -X POST http://localhost:3500/v1.0/publish/myRocketMQ/myTopic?metadata.rock
       }'
 ```
 
+## QueueSelector
+
+The dapr rocketmq component contains five Queue selectors. These are HashQueueSelector, RandomQueueSelector, RoundRobinQueueSelector, ManualQueueSelector, DaprQueueSelector. Four of these are provided by the rocketmq client. DaprQueueSelector is implemented by the dapr rocketmq component. If you want to know more about the Queue selectors hash, roundRobin, Random and Manual, please read https://rocketmq.apache.org/docs.  Let me focus on the design of DaprQueueSelector.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
The RocketMQ component contains a total of five queue selectors. The RocketMQ client provides the following queue selectors:
- `HashQueueSelector`
- `RandomQueueSelector`
- `RoundRobinQueueSelector`
- `ManualQueueSelector`

To learn more about these RocketMQ client queue selectors, read the [RocketMQ documentation](https://rocketmq.apache.org/docs).

The Dapr RocketMQ component implements the following queue selector:
- `DaprQueueSelector`

 This article focuses on the design of `DaprQueueSelector`.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 24]
```diff
@@ -79,7 +101,34 @@ curl -X POST http://localhost:3500/v1.0/publish/myRocketMQ/myTopic?metadata.rock
...
+
+### DaprQueueSelector
+
+DaprQueueSelector integrates three Queue selectors: HashQueueSelector, RoundRobinQueueSelector, ManualQueueSelector.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
`DaprQueueSelector` integrates three queue selectors: 
- `HashQueueSelector`
- `RoundRobinQueueSelector`
- `ManualQueueSelector`
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 25]
```diff
@@ -79,7 +101,34 @@ curl -X POST http://localhost:3500/v1.0/publish/myRocketMQ/myTopic?metadata.rock
...
+
+First, it gets the queue id from the request parameter. You can set the queue id by doing the following.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
`DaprQueueSelector` gets the queue id from the request parameter. You can set the queue id by running the following:
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 26]
```diff
@@ -79,7 +101,34 @@ curl -X POST http://localhost:3500/v1.0/publish/myRocketMQ/myTopic?metadata.rock
...
+
+```
+http://localhost:3500/v1.0/publish/myRocketMQ/myTopic?metadata.rocketmq-queue=1
+```
+
+That's how ManualQueueSelector is implemented. The method above also works for ManualQueueSelector.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
The `ManualQueueSelector` is implemented using the method above.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 27]
```diff
@@ -79,7 +101,34 @@ curl -X POST http://localhost:3500/v1.0/publish/myRocketMQ/myTopic?metadata.rock
...
+
+Second, it will try to get a ShardingKey, and hash the ShardingKey to determine the queue id if it exists.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
Next, the `DaprQueueSelector` tries to:
- Get a `ShardingKey`
- Hash the `ShardingKey` to determine the queue id.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 28]
```diff
@@ -79,7 +101,34 @@ curl -X POST http://localhost:3500/v1.0/publish/myRocketMQ/myTopic?metadata.rock
...
+
+You can set the ShardingKey by doing the following.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
You can set the `ShardingKey` by doing the following:
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md [hunk 29]
```diff
@@ -79,7 +101,34 @@ curl -X POST http://localhost:3500/v1.0/publish/myRocketMQ/myTopic?metadata.rock
...
+
+```
+http://localhost:3500/v1.0/publish/myRocketMQ/myTopic?metadata.rocketmq-shardingkey=key
+```
+
+If the shardingkey does not exist, the RoundRobin algorithm is used to determine the queue id.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-rocketmq.md]:
```suggestion
If the `ShardingKey` does not exist, the `RoundRobin` algorithm is used to determine the queue id.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
Grammar/readability pass

**Comment by jinjianfeng-chn** [comment]
> Grammar/readability pass

Thank you very much for helping me correct the document.  😁

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**